src/mewbot/io/desktop_notification.py

```diff
@@ -0,0 +1,242 @@
+
+    def __init__(self, *args: Optional[Any], **kwargs: Optional[Any]) -> None:
+        self._logger = logging.getLogger(__name__ + "DesktopNotificationIO")
+        # Not entirely sure why, but empty properties in the yaml errors
+        self._logger.info(f"DesktopNotificationIO received args - {args}")
+        self._logger.info(f"DesktopNotificationIO received kwargs - {kwargs}")
+
+    def get_inputs(self) -> Sequence[Input]:
+
+        return []
+
+    def get_outputs(self) -> Sequence[Output]:
+        if not self._output:
+            self._output = DesktopNotificationOutput()
+
+        return [self._output]
+
+
+class DesktopNotificationOutput(Output):
+
+    _engine: DesktopNotificationOutputEngine
+    _logger: logging.Logger
+
+    def __init__(self, *args: Optional[Any], **kwargs: Optional[Any]) -> None:
+        super().__init__(*args, **kwargs)
+
+        self._logger = logging.getLogger(__name__ + "DesktopNotificationOutput")
+        self._logger.info(
+            "Starting DesktopNotificationOutputEngine. Please assume crash positions!"
+        )
+
+        self._engine = DesktopNotificationOutputEngine()
+
+    @staticmethod
+    def consumes_outputs() -> Set[Type[OutputEvent]]:
+        """
+        Defines the set of output events that this Output class can consume
+        :return:
+        """
+        return {DesktopNotificationOutputEvent}
+
+    async def output(self, event: OutputEvent) -> bool:
+        """
+        Does the work of transmitting the event to the world.
+        :param event:
+        :return:
+        """
+        self._logger.info(f"output triggering with {event}")
+
+        if not isinstance(event, DesktopNotificationOutputEvent):
+            return False
+
+        return self._engine.notify(event.title, event.text)
+
+
+class DesktopNotificationOutputEngine:
+    """
+    Contains the dubious and concerning logic required to make output work on all target systems.
+    """
+
+    _logger: logging.Logger
+    _detected_os: str
+    _platform_str: str
+    _enabled: bool
+
+    def __init__(self) -> None:
+
+        self._logger = logging.getLogger(__name__ + "DesktopNotificationOutputEngine")
+        self._logger.info("DesktopNotificationOutputEngine starting")
+
+        # Note-to-self: Warn if we are in a non-tested state
+        # Note-to-self: Need to test to see if we are headless and fallback if we are
+
+        self._platform_str = sys.platform
+        self._logger.info(f"We are detected as running on {self._platform_str}")
+
+        # Could use "in" or string normalisation - but I want it to fail noisily and log the error
+        # if I have made any incorrect assumptions.
+        # So, hopefully, the user will tell us and I can check to see if anything creative
+        # is required.
+        # silent failure of desktop notifications is not desired. Noisy failure is.
+
+        self._enabled = False
+        if self._platform_str == "win32" or self._platform_str == "win64":
```

Comment:
win64 doesn't appear to ever be a valid output from sys.platform
(source: https://docs.python.org/3.10/library/sys.html#sys.platform , not reporting win64 on my version)

Reply:
Removing.

Reply:
(What happens when you go off memory from a decade ago - instead of just reading the docs)